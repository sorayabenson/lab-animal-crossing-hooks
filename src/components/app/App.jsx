import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Header from '../Header';
import Villagers from '../../containers/Villagers';
import Detail from '../../containers/Detail';
import './app.css';

export default function App() {
  return (
    <Router>
      <Header />
      <Switch>
        <Route 
          path="/"
          exact
          render={(routerProps) => <Villagers {...routerProps}/>}/>

        <Route 
          path="/:id"
          render={(routerProps) => <Detail {...routerProps}/>}/>
      </Switch>
    </Router>
  );
}
