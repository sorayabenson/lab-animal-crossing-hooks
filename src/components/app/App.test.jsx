import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';
import { MemoryRouter, Route } from 'react-router-dom';
import mockVillagersData from '../../../fixtures/villagers.json';
import mockAppVillager from '../../../fixtures/app.json';
import { setupServer } from 'msw/node';
import { rest } from 'msw';

const server = setupServer(
  rest.get(
    'https://acnhapi.com/v1/villagers/',
    (req, res, ctx) => {
      return res(
        ctx.json(mockVillagersData)
      )
    }),

  rest.get(
    'https://acnhapi.com/v1/villagers/1',
    (req, res, ctx) => {
      return res(
        ctx.json(mockAppVillager)
      )
    })
)

describe('App component', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('renders App', async () => {
    render(
      <MemoryRouter>
        <Route>
          <App />
        </Route>
      </MemoryRouter>
    );

    const header = screen.getByText(`let's cross some animals!`);
    expect(header).toMatchSnapshot(); 

    //check for villager item for clickability
    const cyrano = await screen.findByText('Cyrano');
    userEvent.click(cyrano);
    const detailCyrano = await screen.findAllByRole('region', { name: 'villager detail' });
    expect(detailCyrano).toMatchSnapshot();
  });
});
