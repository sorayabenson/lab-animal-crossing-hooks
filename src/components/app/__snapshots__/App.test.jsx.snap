// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<h1>
  let's cross some animals!
</h1>
`;
